libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -657,6 +683,14 @@ void CompilerUtils::convertType(Type const& _typeOnStack, Type const& _targetTyp
 
 void CompilerUtils::pushZeroValue(Type const& _type)
 {
+	if (auto const* funType = dynamic_cast<FunctionType const*>(&_type))
+	{
+		if (funType->location() == FunctionType::Location::Internal)
+		{
+			m_context << m_context.errorTag();
```

Comment:
In my taste I would disallow `function() x` without an initialization.  Also, in my taste I would disallow array extension without giving elements, but I realized that's a bigger change and for now we allow array extension without giving elements.